### ISSUE TITLE
Add documentation for nested fields

### ISSUE THREAD

**Issue by mosabua** [trinodb/trino#18546]
for engine and sql language in general (ALTER COLUMN, CREATE TABLE .. and so on) 

And then also for connectors that support this. Currently only Iceberg.

See https://github.com/trinodb/trino/issues/16959 and linked PRs and release notes.

**Comment by mosabua** [comment]
Also see discussion in https://github.com/trinodb/trino/pull/18496 - @ebyhr will help with review

**Comment by dhuynh** [comment]
CREATE TABLE for nested fields exists for JSON - is this what you are looking for?
`CREATE TABLE persons (
    firstName varchar,
    lastName varchar,
    address ROW(
        street varchar,
        city varchar,
        country varchar
    ),
    jobs ARRAY(
        ROW(
            job varchar,
            episode varchar
        )
    )
)
WITH (
   external_location = 's3a://data-lake/persons',
   format = 'JSON'
)`

Does this nested field syntax work across connectors and formats like Hive/Avro and Iceberg/Avro?

**Comment by mosabua** [comment]
Yes .. this is one of the things we need to document .. but we had more features around this in recent releases.
